bundler/lib/bundler/definition.rb

```diff
@@ -825,11 +825,6 @@ def converge_locked_specs
           # commonly happens if the version changed in the gemspec
           next unless new_spec
 
-          new_runtime_deps = new_spec.dependencies.select {|d| d.type != :development }
-          old_runtime_deps = s.dependencies.select {|d| d.type != :development }
-          # If the dependencies of the path source have changed and locked spec can't satisfy new dependencies, unlock it
-          next unless new_runtime_deps.sort == old_runtime_deps.sort || new_runtime_deps.all? {|d| satisfies_locked_spec?(d) }
-
```

Comment:
It fixes https://github.com/ayanko/bundler-gemspec-issue.
Not sure how does it impact on other cases.

Method `converge_locked_specs` is huge and complicated.
Should we split it to more multiple methods to increase clarity?

Reply:
Be my guest!